### PR TITLE
Feature/Routing-Forms/Reporting

### LINKED ISSUE #5440: [CAL-337] Routing Forms - Reporting
Implement a feature to show list of all responses with support of querying over it.

> From [SyncLinear.com](https://synclinear.com)

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Closes #5440

### [Demo](https://www.loom.com/share/b2cd9f83b0ea418a95804c20f90ab090)
Screenshot 2022-11-09 at 9 13 48 PM

**Environment**: Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- See loom

_Note: You would see 2 unit tests check below. One is passing and the other is failing. This is because I have a fix for unit tests workflow in my PR which is passing and `main` one is failing_

### CODE REVIEW

File: packages/app-store/ee/routing-forms/jsonLogicToPrisma.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+export const jsonLogicToPrisma = (jsonLogic) => {
```
Comment (harshsinghatz) [inline packages/app-store/ee/routing-forms/jsonLogicToPrisma.ts:1]: what is the type for jsonLogic?

File: packages/app-store/ee/routing-forms/jsonLogicToPrisma.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  if (!jsonLogic) {
+    return {};
+  }
+  jsonLogic = jsonLogic.logic;
```
Comment (harshsinghatz) [inline packages/app-store/ee/routing-forms/jsonLogicToPrisma.ts:5]: Can we store this in a different variable, as the parameter is directly getting manipulated

File: packages/app-store/ee/routing-forms/jsonLogicToPrisma.ts (hunk 1 above)
Comment (harshsinghatz) [inline packages/app-store/ee/routing-forms/jsonLogicToPrisma.ts:1]: what are the possible types for jsonLogic here?

**Comment by hariombalhara** [comment]
@harshsinghatz this isn't ready for review yet. It's in draft

**Review by hariombalhara — COMMENTED** [review]
Self review done

**Comment by CarinaWolli** [comment]
Awesome feature, works great 🙌🏻 

I only have a few design suggestions: 

- There is more space on the right side of table (I think there is a mr-2 somewhere)
Screenshot 2022-11-10 at 10 11 23

- Pagination would probably make sense to have as there will be a lot of data over time (maybe this can also wait and be added later)
- Should we make the corners of the table rounded to fit our design?

**Review by PeerRich — APPROVED** [review]
the gray headline looks a bit dark, a lighter dark is better

also "Add Query" should be "Add Filter", no?

**Comment by hariombalhara** [comment]
@PeerRich 
- Changed Add Query ->  Add Filter. It would resonate with more people.
- Made header bg-gray-200
Screenshot 2022-11-11 at 11 57 07 AM

@CarinaWolli 
- "There is more space on the right side of table (I think there is a mr-2 somewhere)" -> This is because the table is left aligned. The space at right would vary 
- "Pagination would probably make sense to have as there will be a lot of data over time (maybe this can also wait and be added later)" -> Yeah I thought the same. The user can always filter the data to limit what loads on the page
- "Should we make the corners of the table rounded to fit our design?" - Wanted to do it but it wasn't straightforward.  Tables are tricky. Implemented it now.

**Review by CarinaWolli — APPROVED** [review]

**Pull request merged**